packages/typespec-test/test/Azure.Analytics.Purview.DataMap/spec/routes.tsp

```diff
@@ -0,0 +1,2135 @@
+import "@azure-tools/typespec-azure-core";
+import "@typespec/http";
+import "@typespec/versioning";
+import "@typespec/rest";
+import "./models.tsp";
+
+using TypeSpec.Rest;
+using TypeSpec.Http;
+using TypeSpec.Versioning;
+using Azure.Core;
+using Azure.Core.Traits;
+using Azure.Core.Traits.Private;
+
+namespace PurviewDataMap;
+
+@doc("The ApiVersion query parameter for Atlas API.")
+model AtlasApiVersionParameter {
+  @doc("The API version to use for this operation.")
+  @query("api-version")
+  @minLength(1)
+  apiVersion?: string;
+}
+
+alias AtlasApiVersionParameterTraits = Azure.Core.Traits.VersionParameterTrait<AtlasApiVersionParameter>;
+
```

Comment:
A custom optional `apiVersion` was defined, but since all operations use `AtlasApiVersionParameterTraits`, `optional` is set to `true`.